### ISSUE TITLE
setting the fields description if you are using functional views

### ISSUE THREAD

**Issue by Dean-Christian-Armada** [ekonstantinidis/django-rest-framework-docs#119]
How do I set the field descriptions in views.py on functional views?

**This is my syntax:**
@api_view(['GET', 'POST'])
def api_section(request):
	"""
		GET request gets the section and its details..... POST request, adds a new section
	"""
	if request.method == 'GET':
		sections = Section.objects.all()
		serializer = SectionSerializer(sections, many=True)
		return Response(serializer.data)

	elif request.method == 'POST':
		serializer = SectionSerializer(data=request.data)
		if serializer.is_valid():
			serializer.save()
			serializer_data = serializer.data
			del serializer_data['id']
			return Response(serializer_data, status=status.HTTP_201_CREATED)
		else:
			return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)

**Comment by jsangilve** [comment]
@Dean-Christian-Armada Did you solve this?

**Comment by Dean-Christian-Armada** [comment]
Actually no, I switched to class-based views

**Comment by jsangilve** [comment]
Ok. I imagine as there is not a `serializers_class` decorator, neither a property that can be extracted from function based views, it's not possible get the fields description

**Comment by Dean-Christian-Armada** [comment]
Exactly, besides I found out that the class-based view is much better

**Comment by niscp** [comment]
I am working with the DRF and got struck at this point. I am using fuction based views , and I am not getting the details of input parameters in API doc page. Please help on this one , if someone came across it.

**Comment by EssaAlshammri** [comment]
@niscp have you found a solution ?

**Comment by dmussaku** [comment]
So, if you are using a decorator like `@api_view`, the `pattern.callback` in `rest_framework_docs.api_endpoint.py` will not be `api_section` as you might suspect. But the callable `api_view` which apparently doesn't contain a docstring. 

So there is a way to get a docstring out of `api_view` function however. It is written here:
[http://stackoverflow.com/questions/1782843/python-decorator-handling-docstrings](http://stackoverflow.com/questions/1782843/python-decorator-handling-docstrings) 

Basically you need to rewrite `api_view` in that way that it will contain a decorator `functools.wraps()` which will update the attrs of the decorator.
like so:
```
from functools import wraps

def decorator(f):
    @wraps(f)
    def _decorator():
        print 'decorator active'
        f()
    return _decorator

@decorator
def foo():
    '''the magic foo function'''
    print 'this is function foo'

help(foo)
```
